macros/src/main/scala/MacroCompiler.scala

```diff
@@ -834,36 +834,24 @@ object MacroCompiler extends App {
             )
           ))
         )
-        // Append a NoDCEAnnotation to avoid dead code elimination removing the non-parent SRAMs
-        val state = CircuitState(circuit, HighForm, annotations :+ NoDCEAnnotation)
-
-        // Run the compiler.
-        val result = new MacroCompiler().compileAndEmit(state)
-
-        // Write output FIRRTL file.
-        params.get(Firrtl) match {
-          case Some(firrtlFile: String) => {
-            val fileWriter = new FileWriter(new File(firrtlFile))
-            fileWriter.write(result.circuit.serialize)
-            fileWriter.close()
-          }
-          case None =>
-        }
 
-        // Write output Verilog file.
-        params.get(Verilog) match {
-          case Some(verilogFile: String) => {
-            // Open the writer for the output Verilog file.
-            val verilogWriter = new FileWriter(new File(verilogFile))
+        // The actual MacroCompilerTransform basically just generates an input circuit
+        val macroCompilerInput = CircuitState(circuit, HighForm, annotations)
```

Comment:
I would have assumed this form would match the `getLoweringTransforms` origin form from before.

Reply:
It's actually matching the `CircuitState` input form from before.

However, in reality, it should actually match the `inputForm` of `MacroCompilerTransform`, which is `midForm`, so I'll change that.

Reply:
Before and after this PR, the flow goes Input -> MacroCompilerTransform -> LoweringTransforms. This state I mark the CircuitState with is only applied for the duration of the MacroCompilerTransform, since it gets serialized after, so it just has to agree with that one transform.